vault/auth.go

```diff
@@ -560,6 +549,26 @@ func (c *Core) newCredentialBackend(
 		return nil, fmt.Errorf("unknown backend type: %s", t)
 	}
 
+	// Set up conf to pass in plugin_name
+	conf := make(map[string]string, len(entry.Options)+1)
+	for k, v := range entry.Options {
```

Comment:
In looking through the code, I was surprised this wasn't happening already.  Glad to see you added this.